lib/ecto/adapters/sql.ex

```diff
@@ -38,19 +38,15 @@ defmodule Ecto.Adapters.SQL do
       ## Types
 
       @doc false
-      def autogenerate(:id), do: nil
-      def autogenerate(:embed_id), do: Ecto.UUID.autogenerate()
-      def autogenerate(:binary_id), do: Ecto.UUID.autogenerate()
+      def autogenerate(:id),        do: nil
+      def autogenerate(:embed_id),  do: Ecto.UUID.generate()
```

Comment:
The problem is that in one place we later loaded the autogenerated value (embeds) and in the other we didn't.
In mongo I need raw value in both places. So we either need to dump in both places (but then we need to have different type for embed_id and regular one for SQL databases), or let the adapter control it.